### PR TITLE
Add description of processing of authorized-by

### PR DESCRIPTION
**Pull request by andrew-draper**
Fixes issue #140

### CODE REVIEW

File: draft-ietf-rats-corim.md [hunk 1]
```diff
@@ -1408,36 +1408,51 @@ Evidence MUST be successfully verified.
...
 {::include cddl/accepted-claims-set.cddl}
 ~~~
 
-Verifiers are not required to use this as their internal state, but for the
+Verifiers are not required to use this as their internal state, for the
 purposes of this document a sample Verifier is discussed which uses this format.
 
-The Accepted Claims Set will be matched against CoMID Reference Values, as per
+The Accepted Claims Set (ACS) contains the actual state of Target Environments (TEs).
+The `evidence-triples` field contains Evidence (from Attestors) and Endorsements
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1424]:
```suggestion
The `evidence-triples` field contains Evidence (from Attesters) and Endorsements
```

File: draft-ietf-rats-corim.md [hunk 2]
```diff
@@ -1447,15 +1462,46 @@ If the merged measurement-value-map contains duplicate codepoints and the
...
+Verifier SHALL set the `authorized-by` field in that evidence to the trusted
+authority keys at the head of each key chain which signed that Evidence. This
+key is often the subject of a self-signed certificate.
+
+If multiple authorities approve the Evidence, for example if multiple key chains
+are available, then the `authorized-by` field SHOULD include the key used by
+each of those authorities.
+
+When adding Endorsements resulting from CoRIM processing (see {{sec-add-to-acs}})
+the Verifier SHALL set the `authorized-by` field in that evidence to the
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1493]:
```suggestion
the Verifier SHALL set the `authorized-by` field in that Evidence to the
```

File: draft-ietf-rats-corim.md [hunk 3]
```diff
@@ -1367,7 +1367,7 @@ This concludes the initialisation phase.
 ## Evidence Collection
 
 In the evidence collection phase the Verifier communicates with attesters to
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1369]:
```suggestion
In the Evidence collection phase the Verifier communicates with attesters to
```

File: draft-ietf-rats-corim.md [hunk 4]
```diff
@@ -1408,36 +1408,51 @@ Evidence MUST be successfully verified.
...
+indicates that the authority named by `measurement-map`/`authorized-by`
+asserts that the actual state of the Target Environment `environment-map`
+has the values in `measurement-map`/`mval`.
+
+In `authorized-by`, authority is represented by cryptographic keys. Authority
+is asserted by digitally signing a Claim using the key. Hence, Claims are
+added to the ACS under the authority of a key.
 
-Each entry in the `measurement-values-map` is a separate piece of evidence
+Each entry in the `measurement-values-map` is Evidence or an Endorsement
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1442]: Please create an issue as to how Endorsement matching instead of Evidence is relevant here.

**Review by henkbirkholz — APPROVED** [review]

**Comment by nedmsmith** [comment]
Build fails. I ran fix-lint but found no obvious issues.

**Review by nedmsmith — COMMENTED** [review]
There are several pending comments that should be resolved before accepting the PR.

File: draft-ietf-rats-corim.md [hunk 5]
```diff
@@ -1518,33 +1564,43 @@ Claims Set.
...
 A Verifier SHOULD convert `environment-map` into a form which meets CBOR Core
 Deterministic Encoding Requirements {{-cbor}} before performing the binary comparison.
 
-The Verifier SHALL iterate over the entries in the `measurement-values-map`
-entry within the Reference Value `measurement-map`. Each entry is compared
-against the `measurement-map` from the Accepted Claims Set. If any entry
-does not match then the Reference Value does not match.
+If the Reference Value contains an `authorized-by` field then the Verifier
+SHALL discard any candidate claims whose `authorized-by` field does
+not contain one of the keys listed in the Reference Value `authorized-by`
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1583]:
```suggestion
If the Reference Value claims under the authority of the RVP match claims under the authority of the Attester then the Verifier
SHALL track these claims in the ACS under the authority of both asserters.
If the Verifier has an exclusive appraisal policy that specifies an authority to trust for these claims, the untrusted authorities SHOULD be removed.
If removal of an authority results in claims having no authority, the claims SHOULD be removed.
See the `authorized-by`
```

File: draft-ietf-rats-corim.md [hunk 6]
```diff
@@ -1518,33 +1564,43 @@ Claims Set.
...
+field (see {{sec-authorized-by}} for more details).
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1584]:
```suggestion
field {{sec-authorized-by}} for more details.
```

**Review by nedmsmith — COMMENTED** [review]
added wording describing conditions for when claims can be removed from ACS.

File: draft-ietf-rats-corim.md [hunk 7]
```diff
@@ -1447,15 +1462,46 @@ If the merged measurement-value-map contains duplicate codepoints and the
...
+of competence. Correct processing of the `authorized-by` field lets the Verifier
+filter out claims from Attesters or Endorsers which do not meet its policies.
+
+When adding cryptograpically signed Evidence to the Accepted Claims Set, the
+Verifier SHALL set the `authorized-by` field in that Evidence to the trusted
+authority keys at the head of each key chain which signed that Evidence. This
+key is often the subject of a self-signed certificate.
+
+If multiple authorities approve the Evidence, for example if multiple key chains
+are available, then the `authorized-by` field SHOULD include the key used by
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1489]:
```suggestion
are available, then the ACS `authorized-by` field MUST be updated to include the key used by
```

**Review by nedmsmith — COMMENTED** [review]
added additional comments

File: draft-ietf-rats-corim.md [hunk 8]
```diff
@@ -1408,36 +1408,51 @@ Evidence MUST be successfully verified.
...
 {::include cddl/accepted-claims-set.cddl}
 ~~~
 
-Verifiers are not required to use this as their internal state, but for the
+Verifiers are not required to use this as their internal state, for the
 purposes of this document a sample Verifier is discussed which uses this format.
 
-The Accepted Claims Set will be matched against CoMID Reference Values, as per
+The Accepted Claims Set (ACS) contains the actual state of Target Environments (TEs).
+The `evidence-triples` field contains Evidence (from Attesters) and Endorsements
```
[the remark this answers is not in the document]
Comment (andrew-draper) [reply draft-ietf-rats-corim.md:1424]: I have changed it to state-triples. I think actual-state-triples is too long.

File: draft-ietf-rats-corim.md [hunk 9]
```diff
@@ -1408,36 +1408,51 @@ Evidence MUST be successfully verified.
...
 This document describes an example evidence structure which can be easily
 matched against these Reference Values.
 
-Each set of evidence contains an `environment-map` providing a namespace, and
-a non empty `measurement-values-map`.
+Each entry within `evidence-triples` uses the syntax of `endorsed-triple-record`.
+When an `endorsed-triple-record` appears within `evidence-triples` it
+indicates that the authority named by `measurement-map`/`authorized-by`
+asserts that the actual state of the Target Environment `environment-map`
+has the values in `measurement-map`/`mval`.
```
[the remark this answers is not in the document]
Comment (andrew-draper) [reply draft-ietf-rats-corim.md:1436]: Reworded

File: draft-ietf-rats-corim.md (hunk 4 above)
Comment (andrew-draper) [reply draft-ietf-rats-corim.md:1442]: Can you file the issue? I am not sure what you would like to see changed.

[the remark this answers is not in the document]
Comment (andrew-draper) [reply draft-ietf-rats-corim.md:1442]: We haven't defined EMT yet (that is in the PR #145). I think part of the purpose of this paragraph has been replaced by the introduction 2 paras above. This one and the next are talking about multiple measurements in the same environment etc. I will rephrase.

File: draft-ietf-rats-corim.md [hunk 10]
```diff
@@ -1408,36 +1408,51 @@ Evidence MUST be successfully verified.
...
- these approaches.
+these approaches.
+
+There can be multiple entries in `evidence-triples` which have the
+same `environment-map` and a different `authorized-by` field (see {{sec-authorized-by}}).
 
-If evidence from different sources has the same `environment-map` then the
-`measurement-values-map`s are merged.
+If Evidence or Endorsements from different sources has the same `environment-map`
+and `authorized-by` then the `measurement-values-map`s are merged.
```
[the remark this answers is not in the document]
Comment (andrew-draper) [reply draft-ietf-rats-corim.md:1455]: Text added

File: draft-ietf-rats-corim.md [hunk 11]
```diff
@@ -1408,36 +1408,51 @@ Evidence MUST be successfully verified.
...
 the `environment-map`.
 
 An Attester can provide multiple `environment-map`s each containing a
-`measurement-values-map` with one entry;  a single `environment-map` containing
+`measurement-values-map` with one entry; a single `environment-map` containing
 multiple entries in its `measurement-values-map`; or a combination of
- these approaches.
+these approaches.
+
+There can be multiple entries in `evidence-triples` which have the
```
[the remark this answers is not in the document]
Comment (andrew-draper) [reply draft-ietf-rats-corim.md:1451]: I think this text should go elsewhere. It is not describing ACS

File: draft-ietf-rats-corim.md [hunk 12]
```diff
@@ -1447,15 +1462,46 @@ If the merged measurement-value-map contains duplicate codepoints and the
 measurement values are not equivalent then the verifier SHALL report
 an error and stop validation processing.
 
-### Accepted Claims Set Initialization
+#### Accepted Claims Set Initialization
```
[the remark this answers is not in the document]
Comment (andrew-draper) [reply draft-ietf-rats-corim.md:1465]: ?

File: draft-ietf-rats-corim.md [hunk 13]
```diff
@@ -1456,6 +1465,36 @@ If format translation is required, a CoRIM profile, see {{sec-corim-profile-type
 
 {{sec-dice-spdm}} provides information on how DICE and SPDM Evidence is reformatted into CoMID schema compliant expressions before being added to the Accepted Claims Set.
 
+#### The authorized-by field in Accepted Claims Set {#sec-authorized-by}
+
+The `authorized-by` field in an Accepted Claims Set entry indicates the entity
+whose authority backs the claim.
+An attacker may add entries to the Accepted Claims Set which are outside its area
```
[the remark this answers is not in the document]
Comment (andrew-draper) [reply draft-ietf-rats-corim.md:1479]: Thanks for the improved text. Replaced.

File: draft-ietf-rats-corim.md [hunk 14]
```diff
@@ -1456,6 +1465,36 @@ If format translation is required, a CoRIM profile, see {{sec-corim-profile-type
...
+of competence. Correct processing of the `authorized-by` field lets the Verifier
+filter out claims from Attesters or Endorsers which do not meet its policies.
+
+When adding cryptograpically signed Evidence to the Accepted Claims Set, the
+Verifier SHALL set the `authorized-by` field in that evidence to the trusted
+authority keys at the head of each key chain which signed that Evidence. This
```
[the remark this answers is not in the document]
Comment (andrew-draper) [reply draft-ietf-rats-corim.md:1485]: I think we need to be precise about what is an authority. Can you provide alternative text which tells the verifier exactly what to do?

File: draft-ietf-rats-corim.md (hunk 8 above)
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1424]: #158

Comment (yogeshbdeshpande) [reply draft-ietf-rats-corim.md:1424]: There is some work to be done to fulfil the explanation which will be covered by issue #158

**Review by nedmsmith — APPROVED** [review]
LGTM

**Comment by andrew-draper** [comment]
> added wording describing conditions for when claims can be removed from ACS.

Claims are not removed from ACS. I have updated the wording to try to avoid this interpretation.

File: draft-ietf-rats-corim.md [hunk 15]
```diff
@@ -1577,33 +1632,43 @@ Claims Set.
...
-`reference-values-triple-record` these are packaged together. In other triples multiple
+`reference-triple-record` these are encoded together. In other triples multiple
 Reference Values are represented more compactly by letting one `environment-map`
 apply to multiple `measurement-map`s.
 
 The Verifier first looks for entries in the Accepted Claims Set with the same
-`environment-map` as the Reference Value. If there are no such entries then
-the Reference Value does not match.
+`environment-map` as the Reference Value. These are the candidate claims. If there are
+no candidate claims then the Reference Value does not match.
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1641]:
```suggestion
`environment-map` as the Reference Value. These are the candidate claims. If there are no 
candidate claims then the Reference Value does not match.
```

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

**Review by thomas-fossati — APPROVED** [review]

**Pull request merged**